src/time_integrator/theta_integrator.cpp

```diff
@@ -0,0 +1,134 @@
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_control.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/trilinos_precondition.h>
+
+#include "Wave.hpp"
+
+void ThetaIntegrator::initialize(const TrilinosWrappers::SparseMatrix &M,
+                                 const TrilinosWrappers::SparseMatrix &K,
+                                 const TrilinosWrappers::MPI::Vector  &U0,
+                                 const TrilinosWrappers::MPI::Vector  &V0,
+                                 const double                          dt) {
+    (void) U0;
+    (void) V0;
+
+    // Copy sparsity/layout from M
+    lhs_matrix.reinit(M);
+    rhs_matrix.reinit(M);
+
+    // Build M/dt
+    lhs_matrix.copy_from(M);
+    lhs_matrix *= 1.0 / dt;
+
+    rhs_matrix.copy_from(M);
+    rhs_matrix *= 1.0 / dt;
+
+    // C_LHS = M/dt + theta^2 * dt * K
+    lhs_matrix.add(theta * theta * dt, K);
+
+    // C_RHS^(V) = M/dt - theta(1-theta) * dt * K
+    rhs_matrix.add(-theta * (1.0 - theta) * dt, K);
+}
+
+
+void ThetaIntegrator::advance(const double                                     t_n,
+                              const double                                     dt,
+                              const TrilinosWrappers::SparseMatrix            &M,
+                              const TrilinosWrappers::SparseMatrix            &K,
+                              const TrilinosWrappers::MPI::Vector             &F_n,
+                              const TrilinosWrappers::MPI::Vector             &F_np1,
+                              const AffineConstraints<>                       &constraints_u_np1,
+                              const std::map<types::global_dof_index, double> &u_boundary_values,
+                              const AffineConstraints<>                       &constraints_v_np1,
+                              const std::map<types::global_dof_index, double> &v_boundary_values,
+                              TrilinosWrappers::MPI::Vector                   &U,
+                              TrilinosWrappers::MPI::Vector                   &V)
+{
+    (void)constraints_u_np1;
+    (void)u_boundary_values;
+    (void) t_n;
+    (void) M;
+    (void) K;
+
+    // Save V^n before overwriting it (velocity old)
+    TrilinosWrappers::MPI::Vector V_old(V);
+
+    // Build forcing combo: b = θF^{n+1} + (1-θ)F^n
+    TrilinosWrappers::MPI::Vector rhs(V); // correct partitioning
+    rhs = 0.0;
+    rhs.add(theta, F_np1);
+    rhs.add(1.0 - theta, F_n);
+
+    // Add scheme terms: + C_RHS^(V) * V^n
+    rhs_matrix.vmult_add(rhs, V_old);
+
+    // Add: -K * U^n
+    TrilinosWrappers::MPI::Vector KU(U);
+    K.vmult(KU, U);
+    rhs.add(-1.0, KU);
+
+    const unsigned int max_it = 20000;      // << prima era 5000
+    const double       rel_tol = 1e-8;      // << per output/ParaView va benissimo
+
+    // Solve: C_LHS * V^{n+1} = rhs
+    SolverControl solver_control(max_it, rel_tol);
```

Comment:
This tolerance is not relative, despite what its name suggests! You should consider using the `ReductionControl` class instead.